dataland-frontend/tests/component/components/resources/frameworkDataSearch/ThreeLayerDataTable.cy.ts

```diff
@@ -0,0 +1,147 @@
+
+  /**
+   * Creates a list with the labels of the subcategories inside a cash flow category in the right order.
+   * @param categoryName is the name of the cash flow category
+   * @returns an array with the subcategory labels in the right order
+   */
+  function buildExpectedOrderOfSubcategoriesForCategory(categoryName: "Revenue" | "CapEx" | "OpEx"): string[] {
+    return [
+      `Total Aligned ${categoryName}`,
+      `Total ${categoryName}`,
+      `Total Eligible ${categoryName}`,
+      `Total Non-Aligned ${categoryName}`,
+      `Total Non-Eligible ${categoryName}`,
+    ];
+  }
+
+  const expectedOrderOfSubcategoriesGroupedByCategories: string[][] = [
+    ["Basic Information"],
+    ["Assurance"],
+    buildExpectedOrderOfSubcategoriesForCategory("Revenue"),
+    buildExpectedOrderOfSubcategoriesForCategory("CapEx"),
+    buildExpectedOrderOfSubcategoriesForCategory("OpEx"),
+  ];
+
+  const dataTestTagsOfCashFlowSubcategory = [
+    "totalAlignedShare",
+    "totalAmount",
+    "totalEligibleShare",
+    "totalNonAlignedShare",
+    "totalNonEligibleShare",
+  ];
+
+  const dataTestTagsOfSubcategoriesGroupedByCategories: string[][] = [
+    ["_basicInformation"],
+    ["assurance"],
+    dataTestTagsOfCashFlowSubcategory,
+    dataTestTagsOfCashFlowSubcategory,
+    dataTestTagsOfCashFlowSubcategory,
+  ];
+
+  /**
+   * Toggle a category by clicking on it via its data-test tag
+   * @param dataTestTagOfCategory the data-test tag of the category
+   */
+  function toggleCategoryByClick(dataTestTagOfCategory: string): void {
+    cy.get(`[data-test='${dataTestTagOfCategory}']`).click();
+  }
+
+  it("Check order of the displayed KPIs and its entries", () => {
+    cy.mountWithPlugins(ThreeLayerDataTable, {
+      keycloak: minimalKeycloakMock({}),
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      props: {
+        dataModel: euTaxonomyForNonFinancialsDisplayDataModel,
+        dataAndMetaInfo: mockedDataForTest,
+      },
+    }).then(() => {
+      cy.get("[data-test='TwoLayerTest']")
+        .get(".d-table-style")
+        .each((element, index) => {
+          cy.wrap(element).eq(0).eq(0).get(".p-badge").eq(index).should("have.text", expectedOrderOfCategories[index]);
+        });
+
+      toggleCategoryByClick(dataTestTagsOfCategories[0]);
+
+      /**
+       * The goal for the loop is to expand one KPI at a time and check the order of the entries.
+       */
+
+      let subcategoriesForCurrentCategory;
+      for (let categoryIndex = 1; categoryIndex < dataTestTagsOfCategories.length; categoryIndex++) {
+        subcategoriesForCurrentCategory = dataTestTagsOfSubcategoriesGroupedByCategories[categoryIndex];
+
+        toggleCategoryByClick(dataTestTagsOfCategories[categoryIndex]);
+
+        for (let subCategoryIndex = 0; subCategoryIndex < subcategoriesForCurrentCategory.length; subCategoryIndex++) {
```

Comment:
because the 1st category "basic information" is already opened by default and i wanted to avoid manually closing it before opening it again. anyways, since there is only one entry, the order cant be wrong, so i technicaly dont check it